test/preview/dataclasses/test_document.py

```diff
@@ -31,6 +31,17 @@ def test_document_str(doc, doc_str):
     assert f"Document(id={doc.id}, mimetype: 'text/plain', {doc_str})" == str(doc)
 
 
+def test_init():
```

Comment:
I would like to see two separate test cases here. One should test the init with all defaults, so just `Document()`. A second test should test the init with all custom parameter values. That way we can also test that `self._create_id()` is not called and does not overwrite the custom id provided in the init